tensorflow/core/kernels/uniform_quant_ops/uniform_quantize_op_test.cc

```diff
@@ -22,6 +22,54 @@ namespace tensorflow {
 
 class UniformQuantizeOpsTest : public OpsTestBase {
  protected:
+  struct QParams {
+    QParams(int q_axis, int q_min_val, int q_max_val, std::vector<float> scale,
+            std::vector<int32> zp)
+        : q_axis(q_axis),
+          q_min_val(q_min_val),
+          q_max_val(q_max_val),
+          scale(scale),
+          zp(zp) {}
+    int q_axis;
+    int q_min_val;
+    int q_max_val;
+    std::vector<float> scale;
+    std::vector<int32> zp;
+  };
+
+  template <typename Tin, typename Tout>
+  void TestUniformQuantize(std::vector<Tin> &input, std::vector<long> &shape,
+                           QParams &q_params,
+                           std::vector<Tout> &expected_result) {
+    TF_ASSERT_OK(NodeDefBuilder("test", "UniformQuantize")
+                     .Input(FakeInput(DataTypeToEnum<Tin>::v()))
+                     .Input(FakeInput(DT_FLOAT))
+                     .Input(FakeInput(DT_INT32))
+                     .Attr("Tin", DataTypeToEnum<Tin>::v())
+                     .Attr("Tout", DataTypeToEnum<Tout>::v())
+                     .Attr("quantization_axis", q_params.q_axis)
+                     .Attr("quantization_min_val", q_params.q_min_val)
+                     .Attr("quantization_max_val", q_params.q_max_val)
+                     .Finalize(node_def()));
+    TF_ASSERT_OK(InitOp());
+
+    const long scale_size = static_cast<long>(q_params.scale.size());
+    TensorShape scale_shape(scale_size > 1 ? TensorShape({scale_size})
+                                           : TensorShape({}));
+    const long zp_size = static_cast<long>(q_params.zp.size());
+    TensorShape zp_shape(zp_size > 1 ? TensorShape({zp_size})
+                                     : TensorShape({}));
+    AddInputFromArray<Tin>(TensorShape(shape), input);
+    AddInputFromArray<float>(scale_shape, q_params.scale);
+    AddInputFromArray<int32>(zp_shape, q_params.zp);
+
+    TF_ASSERT_OK(RunOpKernel());
+    Tensor expected(allocator(), DataTypeToEnum<Tout>::v(), TensorShape(shape));
+    // Input element -27.0 is quantized to -127 (not -128) because
+    // output_quantization_min_val is -127.
```

Comment:
I believe this clamping behavior already exists  in the kernel for quantize and is independent of Qint8 and Quint8. With that should we need to highlight in the this test?

Reply:
I agree, I have removed the comment. Thanks!